core-java-modules/core-java-11-2/src/main/java/com/baeldung/httpclient/parameters/HttpClientParametersExample.java

```diff
@@ -0,0 +1,273 @@
+package com.baeldung.httpclient.parameters;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpClientParametersExample {
+
+    public static void main(String[] args) throws Exception {
+        queryParametersRequest();
+
+        noBodyRequests();
+
+        stringBodyRequests();
+
+        byteArrayBodyRequests();
+
+        fileBodyRequest();
+
+        inputStreamBodyRequest();
+    }
+
+    private static void noBodyRequests() throws IOException, InterruptedException {
+        noBodyRequest();
+        noBodyRequestUsingMethod();
+    }
+
+    private static void byteArrayBodyRequests() throws IOException, InterruptedException {
+        byteArrayBodyRequest();
+        byteArrayBodyWithSpecificBytesRequest();
+        byteArraysIteratorRequest();
+    }
+
+    private static void stringBodyRequests() throws IOException, InterruptedException {
+        stringBodyRequest();
+        stringBodyWithCharsetRequest();
+        jsonDataStringBodyRequest();
+        formDataStringBodyRequest();
+    }
+
+    public static void queryParametersRequest() throws IOException, InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
+          .version(HttpClient.Version.HTTP_2)
+          .uri(URI.create("https://postman-echo.com/get?param1=value1&param2=value2"))
+          .GET()
+          .build();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        String responseBody = response.body();
+        int responseStatusCode = response.statusCode();
+
+        System.out.println("Response body : " + responseBody);
+        System.out.println("Response status code: " + responseStatusCode);
+    }
```

Comment:
These four lines are being repeated over and over, so maybe they can be extracted to a method.

Reply:
@JavierPeris Common code has been extracted to a method.